docs/specification/draft/client/elicitation.mdx

```diff
@@ -321,12 +526,212 @@ Servers should handle each state appropriately:
-5. Clients **SHOULD** allow users to decline elicitation requests at any time
-6. Clients **SHOULD** implement rate limiting
-7. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+3. Clients **SHOULD** allow users to decline elicitation requests at any time
+4. Clients **SHOULD** implement rate limiting
+5. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+6. Servers **MUST** bind elicitation requests to the user's identity
+
+### Safe URL Handling
+
+Clients implementing url elicitation **MUST** handle URLs carefully to prevent users from unknowingly clicking malicious links.
+
+1. Servers **MUST NOT** include URLs in any message or schema fields as part of a form elicitation request.
+2. Servers **MUST NOT** include URLs in any message or schema fields as part of an url elicitation request, except for the `url` field.
+
+These requirements ensure that client implementations have clear rules about when to present a URL to the user, so that other rules about user consent and SSRF protection (below) can be consistently applied.
+
+### Identifying the User
+
+Servers **MUST NOT** rely on client-provided user identification, as this can be forged.
+Instead, servers **SHOULD** follow [security best practices](../basic/security_best_practices).
+
+Non-normative examples:
+
+- Incorrect: Treat user input like "I am joe@example.com" as authoritative
+- Correct: Rely on the [MCP authorization server](../basic/authorization) to identify the user
+
+### Form Mode Security
+
+1. Servers **MUST NOT** request sensitive information (passwords, API keys, etc.) via form mode
+2. Clients **SHOULD** validate all responses against the provided schema
+3. Servers **SHOULD** validate received data matches the requested schema
+
+### URL Mode Security
+
+Clients **MUST NOT** open a user agent (e.g. browser) without explicit consent from the user.
+
+Clients **MUST** render the URL provided by the server in a browser which does not enable the client or LLM to inspect the content or user inputs.
+For example, on iOS, [SFSafariViewController](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller) is good, but [WkWebView](https://developer.apple.com/documentation/webkit/wkwebview) is not.
+
+#### Server-Side Request Forgery (SSRF)
+
+Since clients facilitate the opening of URLs provided by servers, they **MUST** implement SSRF protections.
+
+The following are recommendations for implementing SSRF protections. Servers that choose to relax these constraints should be aware of the risks.
+
+- Require the `https://` scheme for all url mode URLs (no HTTP, `file://`, etc.)
+- Block requests to internal IP ranges (e.g., `127.0.0.1`, `10.0.0.0/8`, etc.)
+- Clearly render or distinguish Unicode characters (e.g. punycode URLs) to avoid "look-alike" misdirections
```

Comment:
Assuming this is a client restriction? The heading to this section makes it seem like this is for servers.

Reply:
You're right, this is client stuff. The heading was typo'ed, thanks!